### PR TITLE
#159392305 Refactor Login Page Codebase

### PR DESCRIPTION
**Pull request by albert85**
#### What does this PR do?
Refactor login page codebase

#### Description of Task to be completed?
- refactor component as a PureComponent
- refactor login.scss file

#### How should this be manually tested?
* Clone the repository
* Cd to the project folder
* Run `chmod +x docker.sh` on the terminal
* Start the server using the following commands
    * `./docker.sh development` (for development purposes)  
    *    `./docker.sh production` (for production purposes)
* On the browser, navigate to localhost:3000 for development purposes

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#159392305](https://www.pivotaltracker.com/story/show/159392305)

#### Screenshots (if appropriate)
screencapture-localhost-3000-2018-08-08-12_48_52

#### Questions:
N/A

### CODE REVIEW

**Comment by samuelkubai** [comment]
@albert85 have all the steps clearly outlined on your PR on how to test this and use `docker` commands not `yarn` commands on the PR.

File: src/components/text-link/__tests__/TextLink.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
```
Comment (samuelkubai) [inline src/components/text-link/__tests__/TextLink.test.js]: We decided on using enzyme not react-test-renderer

**Comment by samuelkubai** [comment]
@albert85 Please pull from develop and resolve your conflicts.

File: config/setupTests.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-/* eslint-disable */
-import { configure } from 'enzyme';
+import Enzyme, { shallow, render, mount } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
```
Comment (johnmutuma5) [inline config/setupTests.js]: I like the set up. We can also use a a better snapshot serializer to produce more human readable snapshots. 
I like the one from the `enzyme-to-json` module. 

```
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
```

**Pull request merged**